src/main/java/testapp/adapter/http/UserAccountService.java

```diff
@@ -0,0 +1,67 @@
+package testapp.adapter.http;
+
+import com.google.inject.Inject;
+import com.google.inject.name.Named;
+import com.google.sitebricks.At;
+import com.google.sitebricks.client.transport.Json;
+import com.google.sitebricks.headless.Reply;
+import com.google.sitebricks.headless.Request;
+import com.google.sitebricks.http.Get;
+import com.google.sitebricks.http.Post;
+import testapp.core.*;
+
+@At("/v1/accounts")
+public class UserAccountService {
+    private AccountRepository accRepository;
+
+    @Inject
+    public UserAccountService(AccountRepository accRepository) {
+        this.accRepository = accRepository;
+    }
+
+    @Get
+    @At("/:userId/balance")
+    public Reply<?> getUserBalance(@Named("userId") int userId) {
+        int balance;
+        UserAccount account;
+        try {
+            account = accRepository.findUserAccount(userId);
+            balance = account.getTokenBalance();
+        } catch (AccountException | NullPointerException e) {
+            return Reply.saying().badRequest();
+        }
+        return Reply.with(balance).as(Json.class);
+    }
+
+    @Post
+    @At("/:userId/deposit")
+    public Reply<?> deposit(@Named("userId") int userId, Request request) {
+        UserAccount account;
+        String amount = request.param("amount");
+        int value = Integer.valueOf(amount);
+        try {
+            account = accRepository.findUserAccount(userId);
+            account.deposit(value);
```

Comment:
Why don't you move the deposit and withdraw operations in the repository, this way you can log the operations, and if mistakes happen you can check the log and fix them